ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -602,27 +584,39 @@ let on_request :
           Position.compare_inclusion position range = `Inside)
     in
     Ok (Option.map loc ~f:Range.of_loc, state)
-  | Lsp.Client_request.TextDocumentRename req -> rename state req
-  | Lsp.Client_request.TextDocumentFoldingRange req -> folding_range state req
-  | Lsp.Client_request.SignatureHelp _ -> not_supported ()
-  | Lsp.Client_request.ExecuteCommand _ -> not_supported ()
-  | Lsp.Client_request.TextDocumentLinkResolve l -> Fiber.return @@ Ok (l, state)
-  | Lsp.Client_request.TextDocumentLink _ -> Fiber.return @@ Ok (None, state)
-  | Lsp.Client_request.WillSaveWaitUntilTextDocument _ ->
-    Fiber.return @@ Ok (None, state)
-  | Lsp.Client_request.CodeAction params -> code_action rpc params
-  | Lsp.Client_request.CompletionItemResolve compl ->
+  | TextDocumentRename req -> rename state req
+  | TextDocumentFoldingRange req -> folding_range state req
+  | SignatureHelp _ -> not_supported ()
+  | ExecuteCommand _ -> not_supported ()
+  | TextDocumentLinkResolve l -> Fiber.return @@ Ok (l, state)
+  | TextDocumentLink _ -> Fiber.return @@ Ok (None, state)
+  | WillSaveWaitUntilTextDocument _ -> Fiber.return @@ Ok (None, state)
+  | CodeAction params -> code_action rpc params
+  | CompletionItemResolve ({ data; _ } as compl) ->
+    assert (Option.is_some data);
```

Comment:
Might as well do `let data = Option.value_exn data` instead.

Reply:
I do use `Option.value_exn` later, but wanted to document the assumption and thought assert failure would have a more clear stack trace. Removed.